### ISSUE TITLE
Languageserver does not respect `.lintr` file

### ISSUE THREAD

**Issue by Yousa-Mirage** [r-lib/lintr#2985]
I am using lintr in VSCode on my Arch Linux WSL. I have a `~/.lintr` file in the home path like this and it can be read successfully:
```
linters: linters_with_defaults(
  line_length_linter = line_length_linter(120L),
  assignment_linter = NULL,
  quotes_linter = NULL
  )

```
```
r$> read.dcf("~/.lintr")
     linters                                                                                                                     
[1,] "linters_with_defaults(\nline_length_linter = line_length_linter(120L),\nassignment_linter = NULL,\nquotes_linter = NULL\n)"
```
But it doesn't work. There are still warnings in vscode workplace.

<img width="1001" height="204" alt="Image" src="https://github.com/user-attachments/assets/41a5d085-bece-4dba-b654-666285ba7db5" />

But if I run `lintr::lint("test_lint.r")` directly, there are no lint warnings, so maybe my `.lintr` has been found and loaded correctly. Maybe there are some stupid errors I have made but I can't find.

```
r$> lintr::lint("test_lint.r")
ℹ No lints found.
```

The lintr is newest: `r$> packageVersion("lintr")  [1] ‘3.3.0.9000’` and I am using radian. My VSCode is on Windows and connected to WSL.

**Comment by Yousa-Mirage** [comment]
Well, there are other people who have encountered this problem, like https://github.com/REditorSupport/languageserver/issues/700#issuecomment-3598154856. I solved the problem by downgrading lintr to 3.2.0 successfully. I don't know if this is a `languageserver` or `lintr` problem.

**Comment by Yousa-Mirage** [comment]
This problem is due to `r-languageserver`. They have fixed in https://github.com/REditorSupport/languageserver/pull/706. Before `r-languageserver` publish this update, you can solve this problem by these way:
- use `lintr-3.2.0`: `remotes::install_version("lintr", version = "3.2.0")`
- use newer `languageserver` from github pull: `remotes::install_github("REditorSupport/languageserver@pull/706/head")`

So this issue can be closed.
